IPython/html/services/contents/filemanager.py

```diff
@@ -379,9 +403,17 @@ def save(self, model, path=''):
         self.log.debug("Saving %s", os_path)
         try:
             if model['type'] == 'notebook':
-                nb = nbformat.from_dict(model['content'])
-                self.check_and_sign(nb, path)
-                self._save_notebook(os_path, nb)
+                
+                # NEW
+                file_ext = _file_extension(os_path)
+                if file_ext == '.ipynb':
+                    nb = nbformat.from_dict(model['content'])
+                    self.check_and_sign(nb, path)
+                    self._save_notebook(os_path, nb)
+                elif file_ext == '.md':
```

Comment:
Seems like you could get this same functionality by overriding `_save_file` with something like:

```
def _save_file(self, os_path, content, format):
    if file_ext == '.md':
        return do_special_markdown_stuff()
   else:
        return super(FileContentsManager, self)._save_file(os_path, content, format)
```